server/container/util/src/main/java/org/apache/james/util/ReactorUtils.java

```diff
@@ -53,10 +54,11 @@
             Preconditions.checkArgument(!duration.isZero(), "'windowDuration' must be strictly positive");
 
             return flux -> flux
-                .windowTimeout(elements, duration)
+                .window(elements)
                 .zipWith(Flux.interval(DELAY, duration))
                 .flatMap(Tuple2::getT1, elements, elements)
-                .flatMap(operation, elements);
+                .flatMap(operation, elements)
+                .onErrorContinue((e, o) -> LOGGER.error("Error encountered while throttling for {}", o.toString(), e));
```

Comment:
toString() is causing NullPointerException on james-charge-01

Reply:
Good reason to make it implicit then?